pallets/providers/src/types.rs

```diff
@@ -0,0 +1,121 @@
+use codec::{Decode, Encode, FullCodec, HasCompact, MaxEncodedLen};
+use frame_support::traits::fungible::Inspect;
+use frame_support::BoundedVec;
+use frame_support::{
+    pallet_prelude::*,
+    sp_runtime::traits::{AtLeast32BitUnsigned, CheckEqual, MaybeDisplay, SimpleBitOps},
+    traits::fungible,
+};
+use scale_info::prelude::fmt::Debug;
+use scale_info::TypeInfo;
+
+use crate::Config;
+
+/// Structure that has the possible value propositions that a main storage provider can offer (and the runtime is aware of)
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct ValueProposition<T: Config> {
+    pub data_limit: StorageData<T>,
+    pub protocols: BoundedVec<Protocols<T>, MaxProtocols<T>>,
+    // TODO: Add relevant fields here
+}
+
+/// Structure that represents a main storage provider. It holds the amount of data that the MSP is able to store,
+/// the amount of data that it IS storing, its libp2p multiaddress, and its value proposition.
+#[derive(Encode, Decode, MaxEncodedLen, TypeInfo, RuntimeDebugNoBound, PartialEq, Eq, Clone)]
+#[scale_info(skip_type_params(T))]
+pub struct MainStorageProvider<T: Config> {
+    pub total_data: StorageData<T>,
+    pub data_used: StorageData<T>,
+    pub multiaddress: MultiAddress<T>,
+    pub value_prop: ValueProposition<T>,
```

Comment:
Does this mean that a given MSP can only have ONE value proposition? We could consider changing this into a `BoundedVec` of value propositions. Anyway, we'll focus more on MSPs down the line, so it is fine like this for now.

Reply:
Yep, currently it's only one. My guess is you could have multiple value propositions for different tiers and things like that, so it makes sense to have a vec. I'll keep it as is for this first PR merge but I'll change it for the next one